crates/sui-graphql-client/src/lib.rs

```diff
@@ -194,6 +195,38 @@ impl Client {
             .ok_or_else(|| Error::msg("No data in response"))
     }
 
+    // ===========================================================================
+    // Balance API
+    // ===========================================================================
+
+    /// Get the balance of all the coins owned by address for the provided coin type.
+    /// Coin type will default to `0x2::coin::Coin<0x2::sui::SUI>` if not provided.
+    pub async fn balance(
+        &self,
+        address: Address,
+        coin_type: Option<&str>,
+    ) -> Result<Option<u128>, Error> {
```

Comment:
Is balance a u128 or a u64 in the graphql api?

Reply:
It's a ... BigInt :D
```
BigInt

String representation of an arbitrary width, possibly signed integer.
```

Reply:
Maybe then we just return the String and let the user decide how they want to parse it. 
In the CLI we've been using `u128` for balances though.

Reply:
i ask because theres no way to actually have a balance > a u64 today (I think). I'm fine if we use a u128 though, its much better than a string